src/js/renderers.js

```diff
@@ -0,0 +1,49 @@
+var utils = require('./utils');
+
+var gene = function(config) {
```

Comment:
it doesn't strike me as a 'rule' - a rule is more like a configuration in my mind. This is really what's executing the rules, if anything, don't you think?